Gems/AtomLyIntegration/EMotionFXAtom/Code/Tools/EMStudio/AnimViewportRenderer.cpp

```diff
@@ -0,0 +1,291 @@
+
+        AZ::RPI::RPISystemInterface::Get()->UnregisterScene(m_scene);
+        m_scene = nullptr;
+    }
+
+    void AnimViewportRenderer::Reinit()
+    {
+        ReinitActorEntities();
+        ResetEnvironment();
+    }
+
+    void AnimViewportRenderer::ResetEnvironment()
+    {
+        // Reset environment
+        AZ::Transform iblTransform = AZ::Transform::CreateIdentity();
+        AZ::TransformBus::Event(m_iblEntity->GetId(), &AZ::TransformBus::Events::SetLocalTM, iblTransform);
+
+        const AZ::Matrix4x4 rotationMatrix = AZ::Matrix4x4::CreateIdentity();
+        AZ::RPI::ScenePtr scene = AZ::RPI::RPISystemInterface::Get()->GetDefaultScene();
+        auto skyBoxFeatureProcessorInterface = scene->GetFeatureProcessor<AZ::Render::SkyBoxFeatureProcessorInterface>();
+        skyBoxFeatureProcessorInterface->SetCubemapRotationMatrix(rotationMatrix);
+    }
+
+    void AnimViewportRenderer::ReinitActorEntities()
+    {
+        // 1. Destroy all the entities that do not point to any actorAsset anymore.
+        AZStd::set<AZ::Data::AssetId> assetLookup;
+        AzFramework::EntityContext* entityContext = m_entityContext.get();
+        const size_t numActors = EMotionFX::GetActorManager().GetNumActors();
+        for (size_t i = 0; i < numActors; ++i)
+        {
+            assetLookup.emplace(EMotionFX::GetActorManager().GetActorAsset(i).GetId());
+        }
+        m_actorEntities.erase(
+            AZStd::remove_if(
+                m_actorEntities.begin(), m_actorEntities.end(),
+                [&assetLookup, entityContext](AZ::Entity* entity)
+                {
+                    EMotionFX::Integration::ActorComponent* actorComponent =
+                        entity->FindComponent<EMotionFX::Integration::ActorComponent>();
+                    if (assetLookup.find(actorComponent->GetActorAsset().GetId()) == assetLookup.end())
+                    {
+                        entityContext->DestroyEntity(entity);
+                        return true;
+                    }
+                    return false;
+                }),
+            m_actorEntities.end());
+
+        // 2. Create an entity for every actorAsset stored in actor manager.
+        for (size_t i = 0; i < numActors; ++i)
+        {
+            AZ::Data::Asset<EMotionFX::Integration::ActorAsset> actorAsset = EMotionFX::GetActorManager().GetActorAsset(i);
+            if (!actorAsset->IsReady())
+            {
+                continue;
+            }
+
+            AZ::Entity* entity = FindActorEntity(actorAsset);
+            if (!entity)
+            {
+                m_actorEntities.emplace_back(CreateActorEntity(actorAsset));
+            }
+        }
+    }
+
+    AZ::Entity* AnimViewportRenderer::FindActorEntity(AZ::Data::Asset<EMotionFX::Integration::ActorAsset> actorAsset) const
+    {
+        const auto foundEntity = AZStd::find_if(
+            begin(m_actorEntities), end(m_actorEntities),
```

Comment:
suggestion/comment: You may need `using AZStd::begin` and `using AZStd::end` here to ensure the correct begin/end are found (though if this is working fine across all compilers and they're found through ADL it might be fine)